lib/meow_nx/op/crossover.ex

```diff
@@ -32,4 +32,35 @@ defmodule MeowNx.Op.Crossover do
 
     Nx.select(selection, parents, Nx.reverse(parents, axes: [0]))
   end
+
+  def single_point() do
+    %Op{
+      name: "[Nx] Single point crossover",
+      requires_fitness: false,
+      invalidates_fitness: true,
+      impl: fn population, _ctx ->
+        Op.map_genomes(population, fn genomes ->
+          # TODO: make compiler (and generally other options)
+          # configurable globally for the model
+          Nx.Defn.jit(&single_point_impl(&1), [genomes], compiler: EXLA)
+        end)
+      end
+    }
+  end
+
+  defnp single_point_impl(parents, _opts \\ []) do
+    {n, length} = Nx.shape(parents)
+    half_n = transform(n, &div(&1, 2))
+
+    slice_idx = transform(length, fn length -> :rand.uniform(length - 1) end)
+    left = parents[[0..(n - 1), 0..(slice_idx - 1)]]
+    right = parents[[0..(n - 1), slice_idx..(length - 1)]]
```

Comment:
Ideally we would move this into `Utils.slice_at`, but it should take axis as an option and may not be that straightforward, so we can figure it out separately.